### ISSUE TITLE
Failure in Batch job repository DB using H2 with partitioned step (and other DBs, I think PostgreSQL too)

### ISSUE THREAD

**Issue by scottkurz** [OpenLiberty/open-liberty#18662]
**Describe the bug**

When using an embedded H2 database as my "job repository" (i.e. the target of the batchPersistence DB store), and **running a job with a partitioned step**, I get an exception with the key chained exception stack trace: 

> 
> Caused by: org.h2.jdbc.JdbcSQLException: NULL not allowed for column "PARTITIONED"; SQL statement:
> INSERT INTO WLPSTEPTHREADINSTANCE (PARTNUM, STEPNAME, CHECKPOINTDATA, FK_JOBINSTANCEID, FK_LATEST_STEPEXECID, THREADTYPE) VALUES (?, ?, ?, ?, ?, ?) [23502-190]
>         at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
>         at org.h2.message.DbException.get(DbException.java:179)
>         at org.h2.message.DbException.get(DbException.java:155)
>         at org.h2.table.Column.validateConvertUpdateSequence(Column.java:305)
>         at org.h2.table.Table.validateConvertUpdateSequence(Table.java:749)
>         at org.h2.command.dml.Insert.insertRows(Insert.java:151)
>         at org.h2.command.dml.Insert.update(Insert.java:114)
>         at org.h2.command.CommandContainer.update(CommandContainer.java:78)
>         at org.h2.command.Command.executeUpdate(Command.java:253)
>         at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:157)
>         at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:143)
>         at com.ibm.ws.rsadapter.jdbc.WSJdbcPreparedStatement.executeUpdate(WSJdbcPreparedStatement.java:520)

Unlike the other DBs in which this works, EclipseLink does not generate a default value for this column in the generated DDL, and so we try to insert a 'null' value and blow up.

**Diagnostic information:**
 - OpenLiberty Version: 21.0.0.9

 - server.xml configuration (WITHOUT sensitive information like passwords)
 ```xml
<server description="Arquillian Server">
    <featureManager>
        <feature>batch-1.0</feature>
        <feature>localConnector-1.0</feature>
    </featureManager>

    <batchPersistence jobStoreRef="BatchDatabaseStore"/>

    <databaseStore dataSourceRef="batchHB" id="BatchDatabaseStore"/>

    <library id="H2Lib">
        <fileset dir="${server.config.dir}/lib/global" includes="h2*jar"/>
    </library>

    <dataSource id="batchHB" jndiName="jdbc/batch">
        <jdbcDriver libraryRef="H2Lib"/>
        <properties URL="jdbc:h2:mem:RUNTIMEDB;INIT=runscript from '${shared.resource.dir}/batchH2.ddl'"/>
    </dataSource>
```

(with **batchH2.ddl** containing contents:  `CREATE SCHEMA IF NOT EXISTS JBATCH;`)

**Comment by scottkurz** [comment]
Path to a solution might be to change: https://github.com/OpenLiberty/open-liberty/blob/e7a9c1268d0b293315296cdb2344449a35857518/dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/persistence/jpa/TopLevelStepInstanceEntity.java#L52-L53

to: `@Column(name="PARTITIONED", nullable=true)`

**Comment by scottkurz** [comment]
Fixed by #19006
